### PR TITLE
Correct the construction argument shadowing

### PR DESCRIPTION
**Pull request by ctrl-schaff**
Update regex test and and test for verifying that the default builder parser property matches one generated separately from a builder instance

### CODE REVIEW

**Review by newgene — APPROVED** [review]
Looks good to me

File: biothings/web/query/builder.py [hunk 1]
```diff
@@ -168,22 +168,22 @@ def _verify_default_regex_pattern(self, scopes_regexs):
...
-            else:
-                scopes_regexs.append(scopes_regexs.pop(default_regex_index))
-        return scopes_regexs
+            for regex_pattern, regex_fields in scopes_regexs:
+                regex_pattern = re.compile(regex_pattern)
+                if isinstance(regex_fields, str):
+                    regex_fields = [regex_fields]
+
+                if regex_pattern.pattern == default_regex_pattern.pattern and len(regex_fields) == 0:
+                    continue
```
Comment (newgene) [inline biothings/web/query/builder.py:182]: probably worth adding a line of comment to this block?

File: tests/web/query/builder.py [hunk 2]
```diff
@@ -63,6 +63,8 @@ def test_es():
     assert "_id" in query["_source"]["includes"]
     assert "fieldA" in query["_source"]["includes"]
 
+def test_regex_order
+
```
Comment (newgene) [inline tests/web/query/builder.py:67]: is this unfinished?

Comment (ctrl-schaff) [reply tests/web/query/builder.py]: My bad I had moved the test to `tests/web/query/regex` and forgot to push it on Friday

File: biothings/web/query/builder.py [hunk 3]
```diff
@@ -29,18 +29,21 @@ class implementations or not defined.
...
+import re
+from typing import Iterable
 
-import orjson
 from elasticsearch_dsl import MultiSearch, Q, Search
 from elasticsearch_dsl.exceptions import IllegalOperation
+import orjson
 
 from biothings.utils.common import dotdict
+from biothings.web.settings.default import ANNOTATION_DEFAULT_REGEX_PATTERN
```
Comment (newgene) [inline biothings/web/query/builder.py:46]: importing directly from `default` can be a problem, as it does not allow users to override its value in `config.py` or `config_web.py`. You might want to check `namespace.py` to see how config parameters are passed, e.g. `ANNOTATION_ID_REGEX_LIST`.

File: biothings/web/query/builder.py [hunk 4]
```diff
@@ -140,6 +146,54 @@ def __init__(
...
+        expect: List[re.Pattern, Iterable]
+        3) We then iterate over the structure looking for the default regex pattern.
+            - If we find the default regex pattern match, we ignore updating our list
+            - If we don't find the default regex pattern match, we update our list with the pattern
+        4) At the end we add the default regex pattern because we've exhausted our search of the
+        current pattern list and trimmed any instances we found. This should ensure we've set the
+        default as the last instance in the regex pattern list
+        """
+        default_regex_pattern = ANNOTATION_DEFAULT_REGEX_PATTERN[0]
+        default_regex_fields = ANNOTATION_DEFAULT_REGEX_PATTERN[1]
```
Comment (newgene) [inline biothings/web/query/builder.py:176]: we should also allow `ANNOTATION_DEFAULT_REGEX_PATTERN` to be `None` if we want to disable it completely.

And even if it's not `None`, we should add some checks here to see if it's formatted correctly.

File: biothings/web/query/builder.py (hunk 3 above)
Comment (ctrl-schaff) [reply biothings/web/query/builder.py:46]: Gotcha good to know

**Comment by newgene** [comment]
@ctrl-schaff looks good to me, please check the codacy issues above to see if anything should be addressed, then I think good to merge.

**Pull request merged**